### PR TITLE
libjpeg: add transupp.c to res

### LINKED ISSUE #4222: [question] Using libraries that stopped using semver
For example Google test will stop doing releases (https://github.com/google/googletest/issues/3124). What is the best way forward in this case?

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libjpeg/all**

Copy `transupp.c` to `res` + add test

Fixes #4222

@nminkov Please check

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libjpeg/all/test_package/conanfile.py [hunk 1]
```diff
@@ -8,13 +8,14 @@ class TestPackageConan(ConanFile):
 
     def build(self):
         cmake = CMake(self)
+        cmake.verbose = True
```
Comment (mathbunnyru) [inline recipes/libjpeg/all/test_package/conanfile.py:11]: ### 
```suggestion
```

I think normally verbose is not set. Do we need it here?

**Review by mathbunnyru — APPROVED** [review]

File: recipes/libjpeg/all/test_package/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/libjpeg/all/test_package/conanfile.py:11]: No, it isn't needed.
I added it to debug the commands.

File: recipes/libjpeg/all/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -7,3 +7,7 @@ conan_basic_setup()
 add_executable(${PROJECT_NAME} test_package.c)
 target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
 set_property(TARGET ${PROJECT_NAME} PROPERTY C_STANDARD 99)
+
+add_executable(test_transupp test_transupp.c "${CONAN_LIBJPEG_ROOT}/res/transupp.c")
+target_link_libraries(test_transupp  ${CONAN_LIBS})
```
Comment (mathbunnyru) [inline recipes/libjpeg/all/test_package/CMakeLists.txt:12]:
```suggestion
target_link_libraries(test_transupp ${CONAN_LIBS})
```

File: recipes/libjpeg/all/test_package/test_transupp.c [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    jpeg_stdio_src(&src, fin);
+
+    jcopy_markers_setup(&src, JCOPYOPT_ALL);
+
+    (void)jpeg_read_header(&src, TRUE);
+
+    jpeg_transform_info transform = {
+        .transform = JXFORM_ROT_180,
+//        .perfect = TRUE,
```
Comment (mathbunnyru) [inline recipes/libjpeg/all/test_package/test_transupp.c:62]: Why is this commented?

**Review by mathbunnyru — CHANGES_REQUESTED** [review]

File: recipes/libjpeg/all/test_package/test_transupp.c (hunk 3 above)
Comment (madebr) [reply recipes/libjpeg/all/test_package/test_transupp.c:62]: The test doesn't run successfully with it enabled. The line can be removed.

Comment (madebr) [reply recipes/libjpeg/all/test_package/test_transupp.c:62]:
```suggestion
```

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**